samza-core/src/main/scala/org/apache/samza/system/SystemConsumers.scala

```diff
@@ -192,10 +194,22 @@ class SystemConsumers (
 
     metrics.registerSystemStreamPartition(systemStreamPartition)
     unprocessedMessagesBySSP.put(systemStreamPartition, new ArrayDeque[IncomingMessageEnvelope]())
+
+    // Note regarding Startpoints and MessageChooser:
```

Comment:
Do we always want the startpoint to override the current offset? It seems to me like the chooser should not use the current offset when there is a startpoint, because the startpoint might correspond to a different current offset.